File: 2023-2024/sourcecode/mod1/lab2/12RecyclerViewBasicsJava/app/src/main/java/com/example/a12_recyclerviewbasics_java/WordListViewHolder.java
package com.example.a12_recyclerviewbasics_java;

import android.graphics.Color;
import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.a12_recyclerviewbasics_java.databinding.WordListItemBinding;

public class WordListViewHolder extends RecyclerView.ViewHolder {
    private final WordListItemBinding binding;
    public WordListViewHolder(WordListItemBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(String item){
        binding.singleWord.setText(item);
        binding.getRoot().setOnClickListener(view -> {
            String element = binding.singleWord.getText().toString();
            binding.singleWord.setText("Clicked" + element);
            binding.singleWord.setBackgroundColor(Color.CYAN);
        });
    }
}

File: 2023-2024/sourcecode/mod1/lab3/11RecyclerViewBasicsKotlin/app/src/main/java/com/example/a11_recyclerviewbasics_kotlin/WordListViewHolder.kt
package com.example.a11_recyclerviewbasics_kotlin

import android.graphics.Color
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.example.a11_recyclerviewbasics_kotlin.databinding.WordListItemBinding

class WordListViewHolder(private val binding: WordListItemBinding) :
    RecyclerView.ViewHolder(binding.root) {

    fun bind(item: String) {
        binding.singleWord.text = item
        binding.root.setOnClickListener {
            val element = binding.singleWord.text
            binding.singleWord.text = "Clicked $element"
            binding.singleWord.setBackgroundColor(Color.CYAN)
        }
    }
}